seisbench/generate/augmentation.py

```diff
@@ -670,3 +672,194 @@ def __call__(self, state_dict):
+                        deteriorated by noise. Default is 0.5
+    :type probability: float
+    :param scaling_type: Method how to find the relative amplitude of the input array. Either from the absolute maximum
+                         (max) or from the root-mean-square (rms). Default is max.
+    :type scaling_type: str
+    :param metadata_thresholds: Dictionary containing keys from metadata and threshold values as items to avoid
+                                adding noise samples, if e.g. the signal-to-noise ratio is below a certain threshold:
+                                RealNoise(noise_dataset=noise_dataset,
+                                          metadata_thresholds=dict(
+                                               trace_Z_snr_db=10
+                                          ))
+                                In the example above, noise is only added when SNR of signal of the Z-component is equal
+                                or greater than 10.
+                                Default is None
+    :type metadata_thresholds: dict, None
+    """
+
+    def __init__(
+        self,
+        noise_dataset: seisbench.data.base.WaveformDataset,
+        scale: tuple = (0, 1),
+        key: str = "X",
+        probability: float = 0.5,
+        scaling_type: str = "max",
+        metadata_thresholds: (dict, None) = None,
+    ):
+        if isinstance(key, str):
+            self.key = (key, key)
+        else:
+            self.key = key
+
+        if scaling_type.lower() not in ["rms", "max"]:
+            msg = "Argument scaling_type must be either 'rms' or 'max'."
+            raise ValueError(msg)
+
+        self.noise_dataset = noise_dataset
+        self.scale = scale
+        self.probability = probability
+        self.scaling_type = scaling_type.lower()
+        self.noise_generator = seisbench.generate.GenericGenerator(noise_dataset)
+        self.metadata_thresholds = metadata_thresholds
+        self.noise_samples = len(noise_dataset)
+
+    def __call__(self, state_dict):
+        x, metadata = state_dict[self.key[0]]
+
+        if self.key[0] != self.key[1]:
+            # Ensure metadata is not modified inplace unless input and output key are anyhow identical
+            metadata = copy.deepcopy(metadata)
+
+        # Check all requirements from metadata thresholds
+        # If value from metadata is below given threshold value, then x is returned without added noise
+        if self.metadata_thresholds:
+            for threshold_key, item in self.metadata_thresholds.items():
+                if metadata[threshold_key] < item:
+                    return x
+
+        # Random draw of 0 and 1 with a probability of p
+        draw = np.random.binomial(n=1, p=self.probability)
+
+        if draw == 0:
+            return x
+        else:
+            # Defining scale for noise amplitude
+            scale = np.random.uniform(*self.scale)
+
+            if self.scaling_type == "max":
```

Comment:
Both of these normalisation factors will be off if the trace has a bias. I think it's necessary to remove the mean before. How about:
`np.max(np.abs(x - np.mean(x, axis=-1)))` (Also takes into account that the maximum might be negative.)
For the second normalisation, just `np.std(x)` should do the job.